### PR TITLE
Add DateRange filter in MedicationRequest List page

### PR DESCRIPTION
**Pull request by ivchenkodima**

### CODE REVIEW

File: app/common/helpers/filter.js [hunk 1]
```diff
@@ -22,9 +22,11 @@ export const setFilter = (filter, { location, router }) => {
...
-  const filter = Object.keys(query).find(key =>
-    filters.some(({ name }) => name === key)
-  );
+  const filter = filters[0].names
+    ? Object.keys(query).filter(key =>
+        filters[0].names.some(item => item === key)
+      )
+    : Object.keys(query).find(key => filters.some(({ name }) => name === key));
 
   return filter || defaultFilter;
```
Comment (ivchenkodima) [inline app/common/helpers/filter.js]: Need comment here @kiurchv

File: app/common/containers/pages/ClinicDetailPage/index.js [hunk 2]
```diff
@@ -32,17 +32,19 @@ import styles from './styles.scss';
 @withStyles(styles)
 @translate()
 @provideHooks({
-  fetch: ({ dispatch, params: { id } }) => Promise.all([
-    dispatch(fetchClinic(id)),
-  ]),
+  fetch: ({ dispatch, params: { id } }) =>
+    Promise.all([dispatch(fetchClinic(id))])
```
Comment (kiurchv) [inline app/common/containers/pages/ClinicDetailPage/index.js]: Looks like there is no need in `Promise.all`

Comment (ivchenkodima) [reply app/common/containers/pages/ClinicDetailPage/index.js]: yes)

File: app/common/helpers/filter.js (hunk 1 above)
Comment (kiurchv) [reply app/common/helpers/filter.js]: I've rewritten this function:
```js
export const getFilter = ({ location: { query } }, filters) => {
  const [defaultFilter] = filters;

  const filter = filters.find(
    ({ names, name }) =>
      names
        ? names.every(name => query.hasOwnProperty(name))
        : query.hasOwnProperty(name)
  );

  const { name, names } = filter || defaultFilter;

  return names || name;
};
```

Now it looks much better

File: app/common/containers/pages/MedicationRequestsListPage/index.js [hunk 3]
```diff
@@ -62,6 +66,17 @@ const MedicationRequestsListPage = ({
...
     </div>
+    <div>
+      <DateFilterForm
+        initialValues={
+          activeDateFilter.length === 2 && {
+            created_from: location.query[activeDateFilter[0]],
+            created_to: location.query[activeDateFilter[1]]
+          }
+        }
+        onSubmit={values => setFilter(values, { location, router })}
```
Comment (ivchenkodima) [inline app/common/containers/pages/MedicationRequestsListPage/index.js]: @kiurchv need your comments here

Comment (kiurchv) [reply app/common/containers/pages/MedicationRequestsListPage/index.js]: Assuming that we getting filter from the `getFilter` function from previous comment we can set initial values in that way:
```jsx
<DateFilterForm
  initialValues={activeDateFilter.reduce(
    (filter, name) => ({ ...filter, [name]: location.query[name] }),
    {}
  )}
  onSubmit={values => setFilter(values, { location, router })}
/>
```

**Pull request merged**